### PR TITLE
Add Data support for arrays.

### PR DESCRIPTION
**Pull request by derekdreery**
I've just implemented up to size 16, since it's this kind of thing that increases compile times. It would be easy to expand to 32 if people request it.

### CODE REVIEW

**Comment by derekdreery** [comment]
Closes #884

File: druid/src/data.rs [hunk 1]
```diff
@@ -368,3 +368,29 @@ impl Data for piet::Color {
...
+        impl<T: Data> Data for [T; $this] {
+            fn same(&self, other: &Self) -> bool {
+                self.iter().zip(other.iter()).all(|(a, b)| a.same(b))
+            }
+        }
+        impl_Data!($($rest)*);
+    }
+}
+
+impl_Data! { 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 32 64 128 } // etc
```
Comment (cmyr) [inline druid/src/data.rs:384]: I thought just to 16?

File: druid/src/data.rs [hunk 2]
```diff
@@ -368,3 +368,29 @@ impl Data for piet::Color {
         self.as_rgba_u32().same(&other.as_rgba_u32())
     }
 }
+
+macro_rules! impl_Data {
```
Comment (cmyr) [inline druid/src/data.rs:372]: nit, but I would expect this to be all lowercase? Is there prior art for this sort of thing?

**Comment by zaynetro** [comment]
This is definitely a useful addition!

**Comment by derekdreery** [comment]
ready for review again :)

File: druid/src/data.rs (hunk 2 above)
Comment (Zarenor) [reply druid/src/data.rs]: I think prior art in this project is the `impl_value_type` macros in the Env module. The snake-case this has been updated to now matches that,

**Pull request merged**